### PR TITLE
Add book test

### PR DESCRIPTION
**Pull request by mwafrika**
- Add test
- Add book
- Preserve book data

### CODE REVIEW

File: classes/item.rb [hunk 1]
```diff
@@ -1,39 +1,40 @@
...
+
+  def initialize(publish_date, archived)
+    @id = Random.rand(1..1000)
+    @publish_date = publish_date
+    @archived = archived
+  end
+
+  def genre=(genre)
+    @genre = genre
+    @genre.items.push(self) unless @genre.items.include?(self)
```
Comment (damdafayton) [inline classes/item.rb:15]: @mwafrika please check this, I guess you intended to do
`genre.items.push`

File: classes/item.rb [hunk 2]
```diff
@@ -1,39 +1,40 @@
...
+  end
+
+  def author=(author)
+    @author = author
+    @author.items.push(self) unless @author.items.include?(self)
+  end
+
+  def label=(label)
+    @label = label
+    @label.items.push(self) unless @label.items.include?(self)
```
Comment (damdafayton) [inline classes/item.rb:25]: `label.items.push()` ?

File: classes/item.rb (hunk 1 above)
Comment (mwafrika) [reply classes/item.rb:56]: Okay. working on that

File: classes/item.rb (hunk 2 above)
Comment (mwafrika) [reply classes/item.rb:66]: yeah. it passed correctly. it's part of associations

**Review by cynthiainga — APPROVED** [review]
Hi @mwafrika 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada:

Cheers and Happy coding!:clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**